packages/walletconnect/src/index.ts

```diff
@@ -10,17 +9,19 @@ export const URI_AVAILABLE = 'URI_AVAILABLE'
 
 type MockWalletConnectProvider = WalletConnectProvider & EventEmitter
 
-function parseChainId(chainId: string | number) {
-  return typeof chainId === 'string' ? Number.parseInt(chainId) : chainId
-}
-
-type WalletConnectOptions = Omit<IWCEthRpcConnectionOptions, 'rpc' | 'infuraId' | 'chainId'> & {
-  rpc: { [chainId: number]: string | string[] }
+/**
+ * Rpc map is a map of chainIds to rpc url(s). If multiple urls are provided, the first one that responds
+ * within a given timeout will be used. 
+ * 
+ * Since WalletConnect does not support multiple urls, we will have to resolve the best url before passing it down.
+ */
+type WalletConnectOptions = Omit<Parameters<typeof WalletConnectProvider.init>[0], 'rpcMap'> & {
+  rpcMap?: { [chainId: number]: string | string[] }
```

Comment:
Do we care more about being closer to WC API or backwards-compatible? Debating between `rpcMap` and `rpc` right now

Reply:
Yes, they changed the name. Honestly, I have mixed feelings about either of these approaches. 

On the one hand, this is `walletconnectoptions` derived from the WalletConnect constructor arguments, so you'd like to stay as close to original parameters as possible. That means we should either keep `rpcMap` or move it out of `options` as they are "options to pass to an underlying walletconnect provider" (ideally, without any modification)

On the other hand, like you mentioned, web3-react is already an abstraction, so this should be just fine to do. 

Thoughts?

Reply:
i think similar comment from me as below - if this is an abstraction across connectors, then i would stick with that. if it's specific to WC, then i think matching the name from WC is nice and also bonus points for accepting the old argument but marking it deprecated

Reply:
FWIW the `network` connector already takes an `urlMap` parameter, so two thoughts here:
- this his not a universal web3-react abstraction, but something connector-specific
- the naming already varies per connector

I think I am leaning towards sticking to `rpcMap` to keep it aligned with other WalletConnect options as well as `urlMap` from Network connector.

We could add a depreciation, sure. We will remove it in a later/future release.

Reply:
This is not web3-react specific abstraction, presence of map is highly connector specific. For example, Network connector has `urlMap`. WalletConnect has `rpcMap`, but others connectors are not guaranteed to have/support one. That said, this is highly specific to WC and I would say supporting whatever WC is doing is easiest and most intuitive for the end users. I shipped this + added deprecation note in the comments + fallback.